network: add several CAKE settings

yuwata opened the pull request:
Closes #21043.

bluca commented on man/systemd.network.xml:
```diff
@@ -3490,6 +3490,15 @@ Token=prefixstable:2002:da8:1::</programlisting></para>
+        <term><varname>MPUBytes=</varname></term>
+        <listitem>
+          <para>Rounds each packet (including overhead) up to a minimum length of the bytes. Takes an
+          unsigned integer in the range from 0 to 256. Defaults to unset and kernel's default is used.
```
we usually do `in the range 0…256` for ranges I think?

bluca commented on man/systemd.network.xml:
```diff
@@ -3490,6 +3490,15 @@ Token=prefixstable:2002:da8:1::</programlisting></para>
+      <varlistentry>
+        <term><varname>MPUBytes=</varname></term>
+        <listitem>
+          <para>Rounds each packet (including overhead) up to a minimum length of the bytes. Takes an
```
`up to the specified minimum length.` ?

bluca commented on man/systemd.network.xml:
```diff
@@ -3590,6 +3590,56 @@ Token=prefixstable:2002:da8:1::</programlisting></para>
+
+          <variablelist>
+            <varlistentry>
+              <term><option>bestefort</option></term>
```
`besteffort`?

bluca commented on man/systemd.network.xml:
```diff
@@ -3640,6 +3640,15 @@ Token=prefixstable:2002:da8:1::</programlisting></para>
+      <varlistentry>
+        <term><varname>FirewallMark=</varname></term>
+        <listitem>
+          <para>Takes an unsigned integer in the range 1–4294967295. When specified, firewall-mark-based
```
`1…4294967295`

bluca reviewed (approved):
lgtm, just some nits on documentation. Code looks fine, can't comment on general functionality as I'm not a QoS expert. For some reason though, I really want some cake now...

yuwata commented:
@bluca Thank you for your review.
- Renamed `DifferentiatedServiceMode=` -> `PriorityQueueingPreset=`
- Added one more setting: `UseRawPacketSize=` 
- Updated the man page.

The change is not small, so dropped the green label. PTAL.

Farom reviewed (commented):
It would be useful, if the FLOW ISOLATION PARAMETERS from tc-cake(8) would be mentioned in the man page.

"Flows are defined only by source address [srchost]"

poettering commented on man/systemd.network.xml:
```diff
@@ -3491,6 +3491,14 @@ Token=prefixstable:2002:da8:1::</programlisting></para>
+      <varlistentry>
+        <term><varname>MPUBytes=</varname></term>
+        <listitem>
+          <para>Rounds each packet (including overhead) up to the specified bytes. Takes an unsigned
```
I'd drop the "unsigned", given that the range you document 0…256 makes very clear it's unsigned, the info is thus redundant.

poettering commented on man/systemd.network.xml:
```diff
@@ -3491,6 +3491,14 @@ Token=prefixstable:2002:da8:1::</programlisting></para>
+        <term><varname>MPUBytes=</varname></term>
+        <listitem>
+          <para>Rounds each packet (including overhead) up to the specified bytes. Takes an unsigned
+          integer in the range 0…256. Defaults to unset and kernel's default is used.</para>
```
hmm, the patch actually implements things so that "0" means "leave kernel default", i.e. it has special meaning. Thus this should document 1…256 here I guess.

poettering commented on src/network/tc/cake.c:
```diff
@@ -199,6 +205,65 @@ int config_parse_cake_overhead(
+                           "Invalid '%s=', ignoring assignment: %s",
+                           lvalue, rvalue);
+                return 0;
+        }
```
probably should refuse v == 0 here, given it has special meaning

poettering commented on man/systemd.network.xml:
```diff
@@ -3643,6 +3643,15 @@ Token=prefixstable:2002:da8:1::</programlisting></para>
+      <varlistentry>
+        <term><varname>FirewallMark=</varname></term>
+        <listitem>
+          <para>Takes an unsigned integer in the range 1…4294967295. When specified, firewall-mark-based
```
as above, i'd drop the "unsigned", it's redundant info given you specify the literal range anyway

poettering commented on src/network/tc/cake.c:
```diff
@@ -520,6 +526,59 @@ int config_parse_cake_priority_queueing_preset(
+                           "Failed to parse '%s=', ignoring assignment: %s",
+                           lvalue, rvalue);
+                return 0;
+        }
```
should probably refuse "0", given it's special

yuwata commented:
@Farom and @poettering Thank you for your review. All suggestions are addressed. Upgrading the green label.
